packages/relay/src/lib/eth.ts

```diff
@@ -504,9 +504,11 @@ export class EthImpl implements Eth {
           }
           if (EthImpl.isArrayNonEmpty(contractResultDetails.state_changes)) {
             // filter the state changes to match slot and return value
-            const stateChange = contractResultDetails.state_changes.find(stateChange => stateChange.slot === slot);
-            result = stateChange.value_written;
-          } 
+            const stateChange = contractResultDetails.state_changes.find(stateChange => stateChange.slot === EthImpl.toHex32Byte(slot));
+            if (stateChange) {
```

Comment:
Update `eth.spec.ts` with coverage for result setting and different paths within `toHex32Byte()`